docs(auth, social): Play Store APIs required for google sign in (#6606)
invertase/react-native-firebase: docs/auth/social-auth.md

Upon successful sign-in, any [`onAuthStateChanged`](/auth/usage#listening-to-authentication-state) listeners will trigger
with the new authentication state of the user.

> If you are testing this feature on an android emulator ensure that the emulate has Google Play Enabled, Installed and Updated.

Reviewer: ```suggestion
If you are testing this feature on an android emulator ensure that the emulate is either the Google APIs or Google Play flavor.
```

Author: ok
